2023/10/23/main.go

```diff
@@ -0,0 +1,16 @@
+// Daily Joel 2023-10-23
+//
+// Defer.
+package main
+
+import "fmt"
+
+func main() {
+	name := ""
+
+	defer func() {
+		fmt.Println(name)
+	}()
```

Comment:
Defer delays the anonymous function so that we assign the now empty string variable name with "name", first. 

An anonymous function cannot be called again and lives within the scope of the function it was declared.

Reply:
Nice! Defer delays the execution until the surrounding function returns. In this case, the function is executed when `main()` returns